lib/code_writer.rb

```diff
@@ -141,7 +141,19 @@ def write_return
   end
 
   def write_function(function_name, num_locals)
-    raise NotImplementedError
+    @out.puts "(#{function_name})\n"
+
+    num_locals.times do
+      @out.puts <<~EOF
+        @0
+        D=A
+        @SP
+        A=M
+        M=D
+        @SP
+        M=M+1
+      EOF
```

Comment:
This is correct, but can you think of a more concise way (in Ruby) to emit this particular chunk of 7 instructions? How would you describe the overall task they perform?